<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to Rbowtie • Rbowtie</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to Rbowtie">
<meta property="og:description" content="Rbowtie">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rbowtie</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.29.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Rbowtie-Overview.html">An introduction to Rbowtie</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Rbowtie-Overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to Rbowtie</h1>
                        <h4 class="author">Michael Stadler</h4>
                        <h4 class="author">Dimos Gaidatzis</h4>
                        <h4 class="author">Anita Lerch</h4>
            
            <h4 class="date">07 September, 2020</h4>
      
      
      <div class="hidden name"><code>Rbowtie-Overview.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> package provides an <strong>R</strong> wrapper around the popular <em>bowtie</em> <span class="citation">(Langmead et al. 2009)</span> short read aligner and around <em>SpliceMap</em> <span class="citation">(Au et al. 2010)</span> a <em>de novo</em> splice junction discovery and alignment tool, which makes use of the <em>bowtie</em> software package.</p>
<p>The package is used by the <em><a href="https://bioconductor.org/packages/3.12/QuasR">QuasR</a></em> <span class="citation">(Gaidatzis et al. 2015)</span> bioconductor package to _qu_antify and _a_nnotate _s_hort _r_eads. We recommend to use the <em><a href="https://bioconductor.org/packages/3.12/QuasR">QuasR</a></em> package instead of using <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> directly. The <em><a href="https://bioconductor.org/packages/3.12/QuasR">QuasR</a></em> package provides a simpler interface than <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> and covers the whole analysis workflow of typical ultra-high throughput sequencing experiments, starting from the raw sequence reads, over pre-processing and alignment, up to quantification.</p>
</div>
<div id="preliminaries" class="section level1">
<h1 class="hasAnchor">
<a href="#preliminaries" class="anchor"></a>Preliminaries</h1>
<div id="citing-rbowtie" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-rbowtie" class="anchor"></a>Citing <em>Rbowtie</em>
</h2>
<p>If you use <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> <span class="citation">(Hahne, Lerch, and Stadler 2012)</span> in your work, you can cite it as follows:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"Rbowtie"</span>)
</pre></div>
<pre><code>## 
## The Rbowtie package contains code from two separate software projects.
## If using bowtie only, it can be cited as Langmead et al. (2009). If
## using also SpliceMap, it can be cited in addition Au et al. (2010). The
## Rbowtie package can be cited using the third reference below:
## 
##   Langmead B, Trapnell C, Pop M, Salzberg SL. Ultrafast and
##   memory-efficient alignment of short DNA sequences to the human
##   genome. Genome Biology 10(3):R25 (2009).
## 
##   Au KF, Jiang H, Lin L, Xing Y, Wong WH. Detection of splice junctions
##   from paired-end RNA-seq data by SpliceMap. Nucleic Acids Research,
##   38(14):4570-8 (2010).
## 
##   Hahne F, Lerch A, Stadler MB. Rbowtie: An R wrapper for bowtie and
##   SpliceMap short read aligners. (unpublished)
## 
## This free open-source software implements academic research by the
## authors and co-workers. If you use it, please support the project by
## citing the appropriate journal articles.
## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,
## bibtex=TRUE)', 'toBibtex(.)', or set
## 'options(citation.bibtex.max=999)'.</code></pre>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> is a package for the <strong>R</strong> computing environment and it is assumed that you have already installed <strong>R</strong>. See the <strong>R</strong> project at (<a href="http://www.r-project.org" class="uri">http://www.r-project.org</a>). To install the latest version of <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em>, you will need to be using the latest version of <strong>R</strong>. <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> is part of the Bioconductor project at (<a href="http://www.bioconductor.org" class="uri">http://www.bioconductor.org</a>). To get <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> together with its dependencies you can use</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="st">"BiocManager"</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"Rbowtie"</span>)
</pre></div>
</div>
<div id="loading-of-rbowtie" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-of-rbowtie" class="anchor"></a>Loading of <em>Rbowtie</em>
</h2>
<p>In order to run the code examples in this vignette, the <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> library need to be loaded.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">Rbowtie</span>)
</pre></div>
</div>
<div id="how-to-get-help" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-get-help" class="anchor"></a>How to get help</h2>
<p>Most questions about <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> will hopefully be answered by the documentation or references. If you’ve run into a question which isn’t addressed by the documentation, or you’ve found a conflict between the documentation and software itself, then there is an active support community which can offer help.</p>
<p>The authors of the package (maintainer: Michael Stadler <a href="mailto:michael.stadler@fmi.ch" class="email">michael.stadler@fmi.ch</a>) always appreciate receiving reports of bugs in the package functions or in the documentation. The same goes for well-considered suggestions for improvements.</p>
<p>Any other questions or problems concerning <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> should be posted to the Bioconductor support site (<a href="https://support.bioconductor.org" class="uri">https://support.bioconductor.org</a>). Users posting to the support site for the first time should read the helpful posting guide at (<a href="https://support.bioconductor.org/info/faq/" class="uri">https://support.bioconductor.org/info/faq/</a>). Note that each function in <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> has it’s own help page, e.g. <code><a href="../reference/bowtie_build.html">help("bowtie")</a></code>. Posting etiquette requires that you read the relevant help page carefully before posting a problem to the site.</p>
</div>
</div>
<div id="example-usage-for-individual-rbowtie-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#example-usage-for-individual-rbowtie-functions" class="anchor"></a>Example usage for individual Rbowtie functions</h1>
<p>Please refer to the <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> reference manual or the function documentation (e.g. using <code><a href="../reference/bowtie_build.html">?bowtie</a></code>) for a complete description of <em><a href="https://bioconductor.org/packages/3.12/Rbowtie">Rbowtie</a></em> functions. The descriptions provided below are meant to give and overview over all functions and summarize the purpose of each one.</p>
<div id="build-the-reference-index-with-bowtie_build" class="section level2">
<h2 class="hasAnchor">
<a href="#build-the-reference-index-with-bowtie_build" class="anchor"></a>Build the reference index with <code id="bowtieBuild">bowtie_build</code>
</h2>
<p>To be able to align short reads to a genome, an index has to be build first using the function <code>bowtie_build</code>. Information about arguments can be found with the help of the <code>bowtie_build_usage</code> function or in the manual page <code><a href="../reference/bowtie_build.html">?bowtie_build</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="fu"><a href="../reference/bowtie_build.html">bowtie_build_usage</a></span>()
</pre></div>
<pre><code>##  [1] "Usage: bowtie2-build-s [options]* &lt;reference_in&gt; &lt;ebwt_outfile_base&gt;"            
##  [2] "    reference_in            comma-separated list of files with ref sequences"    
##  [3] "    ebwt_outfile_base       write Ebwt data to files with this dir/basename"     
##  [4] "Options:"                                                                        
##  [5] "    -f                      reference files are Fasta (default)"                 
##  [6] "    -c                      reference sequences given on cmd line (as &lt;seq_in&gt;)" 
##  [7] "    -C/--color              build a colorspace index"                            
##  [8] "    -a/--noauto             disable automatic -p/--bmax/--dcv memory-fitting"    
##  [9] "    -p/--packed             use packed strings internally; slower, uses less mem"
## [10] "    --bmax &lt;int&gt;            max bucket sz for blockwise suffix-array builder"    
## [11] "    --bmaxdivn &lt;int&gt;        max bucket sz as divisor of ref len (default: 4)"    
## [12] "    --dcv &lt;int&gt;             diff-cover period for blockwise (default: 1024)"     
## [13] "    --nodc                  disable diff-cover (algorithm becomes quadratic)"    
## [14] "    -r/--noref              don't build .3/.4.ebwt (packed reference) portion"   
## [15] "    -3/--justref            just build .3/.4.ebwt (packed reference) portion"    
## [16] "    -o/--offrate &lt;int&gt;      SA is sampled every 2^offRate BWT chars (default: 5)"
## [17] "    -t/--ftabchars &lt;int&gt;    # of chars consumed in initial lookup (default: 10)" 
## [18] "    --threads &lt;int&gt;         # of threads"                                        
## [19] "    --ntoa                  convert Ns in reference to As"                       
## [20] "    --seed &lt;int&gt;            seed for random number generator"                    
## [21] "    -q/--quiet              verbose output (for debugging)"                      
## [22] "    -h/--help               print detailed description of tool and its options"  
## [23] "    --usage                 print this usage message"                            
## [24] "    --version               print version information and quit"</code></pre>
<p><code>refFiles</code> below is a vector with filenames of the reference sequence in <code>FASTA</code> format, and <code>indexDir</code> specifies an output directory for the index files that will be generated when calling <code>bowtie_build</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw">refFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(package=<span class="st">"Rbowtie"</span>, <span class="st">"samples"</span>, <span class="st">"refs"</span>), full=<span class="fl">TRUE</span>)
<span class="kw">indexDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"refsIndex"</span>)

<span class="kw">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bowtie_build.html">bowtie_build</a></span>(references=<span class="kw">refFiles</span>, outdir=<span class="kw">indexDir</span>, prefix=<span class="st">"index"</span>, force=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="kw">tmp</span>)
</pre></div>
<pre><code>## [1] "Settings:"                                                                                              
## [2] "  Output files: \"/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpT55irj/refsIndex/index.*.ebwt\""
## [3] "  Line rate: 6 (line is 64 bytes)"                                                                      
## [4] "  Lines per side: 1 (side is 64 bytes)"                                                                 
## [5] "  Offset rate: 5 (one in 32)"                                                                           
## [6] "  FTable chars: 10"</code></pre>
</div>
<div id="create-alignment-with-bowtie" class="section level2">
<h2 class="hasAnchor">
<a href="#create-alignment-with-bowtie" class="anchor"></a>Create alignment with <code>bowtie</code>
</h2>
<p>Information about the arguments supported by the <code>bowtie</code> function can be obtained with the help of the <code>bowtie_usage</code> function or in the manual page <code><a href="../reference/bowtie_build.html">?bowtie</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="../reference/bowtie_build.html">bowtie_usage</a></span>()
</pre></div>
<pre><code>##  [1] "Usage: "                                                                                        
##  [2] "bowtie-align-s [options]* &lt;ebwt&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | --12 &lt;r&gt; | --interleaved &lt;i&gt; | &lt;s&gt;} [&lt;hit&gt;]"
##  [3] ""                                                                                               
##  [4] "  &lt;m1&gt;    Comma-separated list of files containing upstream mates (or the"                      
##  [5] "          sequences themselves, if -c is set) paired with mates in &lt;m2&gt;"                        
##  [6] "  &lt;m2&gt;    Comma-separated list of files containing downstream mates (or the"                    
##  [7] "          sequences themselves if -c is set) paired with mates in &lt;m1&gt;"                         
##  [8] "  &lt;r&gt;     Comma-separated list of files containing Crossbow-style reads.  Can be"               
##  [9] "          a mixture of paired and unpaired.  Specify \"-\" for stdin."                          
## [10] "  &lt;i&gt;     Files with interleaved paired-end FASTQ reads."                                       
## [11] "  &lt;s&gt;     Comma-separated list of files containing unpaired reads, or the"                      
## [12] "          sequences themselves, if -c is set.  Specify \"-\" for stdin."                        
## [13] "  &lt;hit&gt;   File to write hits to (default: stdout)"                                              
## [14] "Input:"                                                                                         
## [15] "  -q                 query input files are FASTQ .fq/.fastq (default)"                          
## [16] "  -f                 query input files are (multi-)FASTA .fa/.mfa"                              
## [17] "  -F k:&lt;int&gt;,i:&lt;int&gt; query input files are continuous FASTA where reads"                        
## [18] "                     are substrings (k-mers) extracted from a FASTA file &lt;s&gt;"                   
## [19] "                     and aligned at offsets 1, 1+i, 1+2i ... end of reference"                  
## [20] "  -r                 query input files are raw one-sequence-per-line"                           
## [21] "  -c                 query sequences given on cmd line (as &lt;mates&gt;, &lt;singles&gt;)"                 
## [22] "  -C                 reads and index are in colorspace"                                         
## [23] "  -Q/--quals &lt;file&gt;  QV file(s) corresponding to CSFASTA inputs; use with -f -C"                
## [24] "  --Q1/--Q2 &lt;file&gt;   same as -Q, but for mate files 1 and 2 respectively"                       
## [25] "  -s/--skip &lt;int&gt;    skip the first &lt;int&gt; reads/pairs in the input"                             
## [26] "  -u/--qupto &lt;int&gt;   stop after first &lt;int&gt; reads/pairs (excl. skipped reads)"                  
## [27] "  -5/--trim5 &lt;int&gt;   trim &lt;int&gt; bases from 5' (left) end of reads"                              
## [28] "  -3/--trim3 &lt;int&gt;   trim &lt;int&gt; bases from 3' (right) end of reads"                             
## [29] "  --phred33-quals    input quals are Phred+33 (default)"                                        
## [30] "  --phred64-quals    input quals are Phred+64 (same as --solexa1.3-quals)"                      
## [31] "  --solexa-quals     input quals are from GA Pipeline ver. &lt; 1.3"                               
## [32] "  --solexa1.3-quals  input quals are from GA Pipeline ver. &gt;= 1.3"                              
## [33] "  --integer-quals    qualities are given as space-separated integers (not ASCII)"               
## [34] "Alignment:"                                                                                     
## [35] "  -v &lt;int&gt;           report end-to-end hits w/ &lt;=v mismatches; ignore qualities"                
## [36] "    or"                                                                                         
## [37] "  -n/--seedmms &lt;int&gt; max mismatches in seed (can be 0-3, default: -n 2)"                        
## [38] "  -e/--maqerr &lt;int&gt;  max sum of mismatch quals across alignment for -n (def: 70)"               
## [39] "  -l/--seedlen &lt;int&gt; seed length for -n (default: 28)"                                          
## [40] "  --nomaqround       disable Maq-like quality rounding for -n (nearest 10 &lt;= 30)"               
## [41] "  -I/--minins &lt;int&gt;  minimum insert size for paired-end alignment (default: 0)"                 
## [42] "  -X/--maxins &lt;int&gt;  maximum insert size for paired-end alignment (default: 250)"               
## [43] "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (default: --fr)"                  
## [44] "  --nofw/--norc      do not align to forward/reverse-complement reference strand"               
## [45] "  --maxbts &lt;int&gt;     max # backtracks for -n 2/3 (default: 125, 800 for --best)"                
## [46] "  --pairtries &lt;int&gt;  max # attempts to find mate for anchor hit (default: 100)"                 
## [47] "  -y/--tryhard       try hard to find valid alignments, at the expense of speed"                
## [48] "  --chunkmbs &lt;int&gt;   max megabytes of RAM for best-first search frames (def: 64)"               
## [49] " --reads-per-batch   # of reads to read from input file at once (default: 16)"                  
## [50] "Reporting:"                                                                                     
## [51] "  -k &lt;int&gt;           report up to &lt;int&gt; good alignments per read (default: 1)"                  
## [52] "  -a/--all           report all alignments per read (much slower than low -k)"                  
## [53] "  -m &lt;int&gt;           suppress all alignments if &gt; &lt;int&gt; exist (def: no limit)"                  
## [54] "  -M &lt;int&gt;           like -m, but reports 1 random hit (MAPQ=0); requires --best"               
## [55] "  --best             hits guaranteed best stratum; ties broken by quality"                      
## [56] "  --strata           hits in sub-optimal strata aren't reported (requires --best)"              
## [57] "Output:"                                                                                        
## [58] "  -t/--time          print wall-clock time taken by search phases"                              
## [59] "  -B/--offbase &lt;int&gt; leftmost ref offset = &lt;int&gt; in bowtie output (default: 0)"                 
## [60] "  --quiet            print nothing but the alignments"                                          
## [61] "  --refidx           refer to ref. seqs by 0-based index rather than name"                      
## [62] "  --al &lt;fname&gt;       write aligned reads/pairs to file(s) &lt;fname&gt;"                              
## [63] "  --un &lt;fname&gt;       write unaligned reads/pairs to file(s) &lt;fname&gt;"                            
## [64] "  --no-unal          suppress SAM records for unaligned reads"                                  
## [65] "  --max &lt;fname&gt;      write reads/pairs over -m limit to file(s) &lt;fname&gt;"                        
## [66] "  --suppress &lt;cols&gt;  suppresses given columns (comma-delim'ed) in default output"               
## [67] "  --fullref          write entire ref name (default: only up to 1st space)"                     
## [68] "Colorspace:"                                                                                    
## [69] "  --snpphred &lt;int&gt;   Phred penalty for SNP when decoding colorspace (def: 30)"                  
## [70] "     or"                                                                                        
## [71] "  --snpfrac &lt;dec&gt;    approx. fraction of SNP bases (e.g. 0.001); sets --snpphred"               
## [72] "  --col-cseq         print aligned colorspace seqs as colors, not decoded bases"                
## [73] "  --col-cqual        print original colorspace quals, not decoded quals"                        
## [74] "  --col-keepends     keep nucleotides at extreme ends of decoded alignment"                     
## [75] "SAM:"                                                                                           
## [76] "  -S/--sam           write hits in SAM format"                                                  
## [77] "  --mapq &lt;int&gt;       default mapping quality (MAPQ) to print for SAM alignments"                
## [78] "  --sam-nohead       supppress header lines (starting with @) for SAM output"                   
## [79] "  --sam-nosq         supppress @SQ header lines for SAM output"                                 
## [80] "  --sam-RG &lt;text&gt;    add &lt;text&gt; (usually \"lab=value\") to @RG line of SAM header"              
## [81] "Performance:"                                                                                   
## [82] "  -o/--offrate &lt;int&gt; override offrate of index; must be &gt;= index's offrate"                     
## [83] "  -p/--threads &lt;int&gt; number of alignment threads to launch (default: 1)"                        
## [84] "  --mm               use memory-mapped I/O for index; many 'bowtie's can share"                 
## [85] "  --shmem            use shared mem for index; many 'bowtie's can share"                        
## [86] "Other:"                                                                                         
## [87] "  --seed &lt;int&gt;       seed for random number generator"                                          
## [88] "  --verbose          verbose output (for debugging)"                                            
## [89] "  --version          print version information and quit"                                        
## [90] "  -h/--help          print this usage message"</code></pre>
<p>In the example below, <code>readsFiles</code> is the name of a file containing short reads to be aligned with <code>bowtie</code>, and <code>samFiles</code> specifies the name of the output file with the generated alignments.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="kw">readsFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(package=<span class="st">"Rbowtie"</span>, <span class="st">"samples"</span>, <span class="st">"reads"</span>, <span class="st">"reads.fastq"</span>)
<span class="kw">samFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"alignments.sam"</span>)

<span class="fu"><a href="../reference/bowtie_build.html">bowtie</a></span>(sequences=<span class="kw">readsFiles</span>, 
       index=<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">indexDir</span>, <span class="st">"index"</span>), 
       outfile=<span class="kw">samFiles</span>, sam=<span class="fl">TRUE</span>,
       best=<span class="fl">TRUE</span>, force=<span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/strtrim.html">strtrim</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span>(<span class="kw">samFiles</span>), <span class="fl">65</span>)
</pre></div>
<pre><code>## [1] "@HD\tVN:1.0\tSO:unsorted"                                                  
## [2] "@SQ\tSN:chr1\tLN:100000"                                                   
## [3] "@SQ\tSN:chr2\tLN:100000"                                                   
## [4] "@SQ\tSN:chr3\tLN:100000"                                                   
## [5] "@PG\tID:Bowtie\tVN:1.2.3\tCL:\"/Users/runner/work/_temp/Library/Rbowt"     
## [6] "HWUSI-EAS1513_0012:6:48:5769:946#0/1\t0\tchr1\t819\t255\t101M\t*\t0\t0\tTG"
## [7] "HWUSI-EAS1513_0012:6:48:6908:952#0/1\t0\tchr2\t1133\t255\t101M\t*\t0\t0\tA"
## [8] "HWUSI-EAS1513_0012:6:48:8070:953#0/1\t0\tchr1\t7543\t255\t101M\t*\t0\t0\tG"</code></pre>
</div>
<div id="create-spliced-alignment-with-splicemap" class="section level2">
<h2 class="hasAnchor">
<a href="#create-spliced-alignment-with-splicemap" class="anchor"></a>Create spliced alignment with <code>SpliceMap</code>
</h2>
<p>While <code>bowtie</code> only generates ungapped alignments, the <code>SpliceMap</code> function can be used to generate spliced alignments. <code>SpliceMap</code> is itself using <code>bowtie</code>. To use it, it is necessary to create an index of the reference sequence as described in @ref(bowtieBuild). <code>SpliceMap</code> parameters are specified in the form of a named list, which follows closely the configure file format of the original <code>SpliceMap</code> program<span class="citation">(Au et al. 2010)</span>. Be aware that <code>SpliceMap</code> can only be used for reads that are at least 50bp long.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="kw">readsFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(package=<span class="st">"Rbowtie"</span>, <span class="st">"samples"</span>, <span class="st">"reads"</span>, <span class="st">"reads.fastq"</span>)
<span class="kw">refDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(package=<span class="st">"Rbowtie"</span>, <span class="st">"samples"</span>, <span class="st">"refs"</span>, <span class="st">"chr1.fa"</span>)
<span class="kw">indexDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"refsIndex"</span>)
<span class="kw">samFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(), <span class="st">"splicedAlignments.sam"</span>)

<span class="kw">cfg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(genome_dir=<span class="kw">refDir</span>,
            reads_list1=<span class="kw">readsFiles</span>,
            read_format=<span class="st">"FASTQ"</span>,
            quality_format=<span class="st">"phred-33"</span>,
            outfile=<span class="kw">samFiles</span>,
            temp_path=<span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>(),
            max_intron=<span class="fl">400000</span>,
            min_intron=<span class="fl">20000</span>,
            max_multi_hit=<span class="fl">10</span>,
            seed_mismatch=<span class="fl">1</span>,
            read_mismatch=<span class="fl">2</span>,
            num_chromosome_together=<span class="fl">2</span>,
            bowtie_base_dir=<span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="kw">indexDir</span>, <span class="st">"index"</span>),
            num_threads=<span class="fl">4</span>,
            try_hard=<span class="st">"yes"</span>,
            selectSingleHit=<span class="fl">TRUE</span>)
<span class="kw">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpliceMap.html">SpliceMap</a></span>(<span class="kw">cfg</span>)
</pre></div>
<pre><code>## [SpliceMap] splitting reads into 25mers...done
## [SpliceMap] aligning 25mers...done
## [SpliceMap] sorting 25mer-alignments...done
## [SpliceMap] indexing 25mer-alignments...done
## [SpliceMap] finding spliced alignments for 1 chromosomes using 2 parallel processes...done
## [SpliceMap] extracting unmapped reads (single read mode)...done
## [SpliceMap] combining spliced alignments...done
## [SpliceMap] reordering final output...done</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">res</span>
</pre></div>
<pre><code>## [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpT55irj/splicedAlignments.sam"</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/strtrim.html">strtrim</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span>(<span class="kw">samFiles</span>), <span class="fl">65</span>)
</pre></div>
<pre><code>## [1] "@HD\tVN:1.0\tSO:coordinate"                                                
## [2] "@SQ\tSN:chr1\tLN:100000"                                                   
## [3] "@PG\tID:SpliceMap\tVN:3.3.5.2 (55)"                                        
## [4] "HWUSI-EAS1513_0012:6:48:5769:946#0\t0\tchr1\t819\t255\t101M\t*\t0\t0\tTGGA"
## [5] "HWUSI-EAS1513_0012:6:48:6908:952#0\t4\t*\t0\t0\t*\t*\t0\t0\tAACATAGTGAAGAA"
## [6] "HWUSI-EAS1513_0012:6:48:8070:953#0\t0\tchr1\t7543\t255\t101M\t*\t0\t0\tGTC"
## [7] "HWUSI-EAS1513_0012:6:48:9942:949#0\t4\t*\t0\t0\t*\t*\t0\t0\tCGGTTCCTGTATCC"</code></pre>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<p>The output in this vignette was produced under:</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
</pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] Rbowtie_1.29.1   BiocStyle_2.17.0
## 
## loaded via a namespace (and not attached):
##  [1] bookdown_0.20       digest_0.6.25       crayon_1.3.4       
##  [4] rprojroot_1.3-2     assertthat_0.2.1    R6_2.4.1           
##  [7] backports_1.1.9     magrittr_1.5        evaluate_0.14      
## [10] stringi_1.4.6       rlang_0.4.7         fs_1.5.0           
## [13] ragg_0.3.1          rmarkdown_2.3       pkgdown_1.5.1.9000 
## [16] desc_1.2.0          tools_4.0.2         stringr_1.4.0      
## [19] parallel_4.0.2      yaml_2.2.1          xfun_0.16          
## [22] compiler_4.0.2      systemfonts_0.3.0   BiocManager_1.30.10
## [25] cpp11_0.2.1         memoise_1.1.0       htmltools_0.5.0    
## [28] knitr_1.29</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-SpliceMap">
<p>Au, K. F., H. Jiang, L. Lin, Y. Xing, and W. H. Wong. 2010. “Detection of Splice Junctions from Paired-End Rna-Seq Data by Splicemap.” <em>Nucleic Acids Research</em> 38 (14): 4570–8.</p>
</div>
<div id="ref-QuasR">
<p>Gaidatzis, D., A. Lerch, F. Hahne, and M. B. Stadler. 2015. “QuasR: Quantify and Annotate Short Reads in R.” <em>Bioinformatics</em> 31 (7): 1130–2. <a href="https://doi.org/10.1093/bioinformatics/btu781">https://doi.org/10.1093/bioinformatics/btu781</a>.</p>
</div>
<div id="ref-Rbowtie">
<p>Hahne, F., A. Lerch, and M. B. Stadler. 2012. “Rbowtie: A R Wrapper for Bowtie and SpliceMap Short Read Aligners.”</p>
</div>
<div id="ref-bowtie">
<p>Langmead, B., C. Trapnell, M. Pop, and S. L. Salzberg. 2009. “Ultrafast and Memory-Efficient Alignment of Short Dna Sequences to the Human Genome.” <em>Genome Biology</em> 10 (3): R25.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Michael Stadler.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
